### DISCUSSION TITLE
카메라를 이용한 반사효과 질문

### DISCUSSION THREAD

**Discussion by YSHFM** [naverz/zepeto-studio-kor#976] [5. Q&A]
현재 거울에 달려 있는 카메라와 렌더텍스쳐를 이용하여 반사효과를 구현해보고 있습니다.

![image](https://user-images.githubusercontent.com/106574541/177781226-e7266ac0-aa5e-4b2f-a411-2f8f73f5a425.png)

각도는 둘째치고, 유니티에선 캐릭터가 반사에 잡힙니다.

![image](https://user-images.githubusercontent.com/106574541/177781516-8b8a1c70-500b-4a2f-ade8-c5ff293188a2.png)

하지만 모바일에서는 보시는 바와같이 캐릭터 그림자는 비치지만 캐릭터 메쉬만 반사에 잡히지 않습니다. 
갤럭시랑 ios동일한데 이유가 감이 잡히질 않습니다....

고수분들의 도움이 절실합니다 ㅠㅠㅠㅠㅠ

**Comment by Zepeto-TechEvangelist** [comment]
안녕하세요~  렌더텍스쳐 포맷 및 표면 매터리얼에 적용한 셰이더가 어떤것인지 알려주시면 문제 해결에 도움이 될 것 같습니다.
스크린샷으로 봐서는 렌더텍스처용 카메라 옵션에 near plane 값이 너무 여유로워서 캐릭터가 짤린 게 아닐까 의심이 됩니다.

그리고 동일한 각도에서 보았을 때, 동일한 캐릭터를 사용했을때 유니티 에디터와 모바일의 경우도 보여주시면 좋을 것 같습니다.

**Reply by YSHFM** [reply]
확인결과 플레이어 기본쉐이더입니다. 빈 쉐이더 생성 후 반사되는 것을 확인했으나 Z값 정렬이 제대로 이루어지지 않는 것 같습니다. 
#891 해당 이슈로 인해 플레이어의 모든 쉐이더를 변경할 수가 없습니다.(눈 메시 등...)

캐릭터의 문제는 아님을 이미 확인했습니다.

![image](https://user-images.githubusercontent.com/106574541/178170016-9541b7c8-0e16-4642-bcc1-adbb04976a81.png)

![image](https://user-images.githubusercontent.com/106574541/178170080-5a71faf6-2724-4b1f-b8f0-e74ba4536a3e.png)
 위 두 사진 모두 ios환경에서 캡처한 사진이며, 안드로이드에서 동일함을 확인했습니다. 유니티에서는 말씀드렸다시피 정상적으로 출력됩니다. 사진속에서 팔부분을 보시다시피 팔이 옷 메시를 뚫고 나오는 모습도 확인이 됩니다.

culling mask또한 전체로 체크했으며 코드로도 삽입했으나 동일했습니다.

**Reply by Zepeto-TechEvangelist** [reply]
확인 감사합니다. #891 이슈가 처리되려면 시간이 다소 소요될 것 같습니다.
최대한 우선 순위를 높일 수 있도록 조율해 보겠습니다. 
감사합니다.
